lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -267,9 +267,33 @@ JoinOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
   TypeRange leftFieldTypes = cast<TupleType>(leftInput.getType()).getTypes();
   TypeRange rightFieldTypes = cast<TupleType>(rightInput.getType()).getTypes();
 
+  // get join type
+  Adaptor adaptor(operands, attributes, properties, regions);
+
+  JoinTypeKind join_type = adaptor.getJoinType();
+
```

Comment:
Oh, that's elegant! The adaptor is so much nicer than the casting of properties I had done in the past... thanks for teaching me!

Nits: Make comment a full sentence and group the two related statements into a block.

```suggestion
  // Get accessor to `join_type`.
  Adaptor adaptor(operands, attributes, properties, regions);
  JoinTypeKind join_type = adaptor.getJoinType();
```